node/modules/storageminer_idxprov.go

```diff
@@ -93,13 +93,25 @@ func IndexProvider(cfg config.IndexProviderConfig) func(params IdxProv, marketHo
 			// The extra data is required by the lotus-specific index-provider gossip message validators.
 			ma := address.Address(maddr)
 			opts = append(opts,
-				engine.WithPublisherKind(engine.DataTransferPublisher),
 				engine.WithDataTransfer(dtV1ToIndexerDT(dt, func() ipld.LinkSystem {
```

Comment:
I think we won't need DT config nor gossipsub config if HTTP publisher kind is configured. @LexLuthr @dirkmc can I ask you to give this a quick test just to make sure please?

And if all is good, we can move those two options into the if clause below.